terraform/modules/cluster/gke-beta/scripts/gke_cluster.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+gke_cluster::create () {
+    gcloud container clusters describe ${cluster_name} --region ${region} \
+    || { gcloud beta container clusters create ${cluster_name} --region ${region} \
+      --project ${project_id} \
+      --cluster-version ${version} \
+      --enable-ip-alias \
+      --enable-multi-networking \
+      --enable-dataplane-v2 \
+      --workload-pool=${project_id}.svc.id.goog \
+      --num-nodes 1 \
```

Comment:
is `1` the number of nodes in default NodePool that we then delete? Can you specify `0` here to avoid creation in the first place?

Reply:
agree with Dmitry. I looked at the guide from GKE, we don't need to create a default node pool: https://docs.google.com/document/d/1tCQBzv97KLYxAqCuI88Do38RAef7rLJJu0GL1_QjWTQ/edit#bookmark=id.l2lrf3831cg5

From the guide: 
gcloud beta container clusters create ${CLUSTER_NAME} \
--enable-dataplane-v2 \
--zone ${ZONE} \
--enable-ip-alias \
--enable-multi-networking \
--cluster-version ${GKE_VERSION} \
--project ${PROJECT} \
--impersonate-service-account "${SA_FULL_NAME?}"

so, let's just drop the --num-nodes at all.

Reply:
If we drop the `--num-nodes` it is defaulted to 3. So the behavior of GKE is they always create a default pool and then you have to call a `gcloud containers node-pool delete` command to delete it. They have a flag in terraform `remove_default_node_pool` to delete the default pool automatically though. 

But I think setting `--num-nodes` to 0 is a good idea. let me try this and if it works we can do that.

Reply:
i am getting below error
```
ERROR: (gcloud.beta.container.clusters.create) argument --num-nodes: Value must be greater than or equal to 1; received: 0
```

So we have to set it to 1.